### PR TITLE
docs: add island integration guide

### PR DESCRIPTION
**Pull request by ericpaulsen**
this PR adds a guide for our native integration with [island.io](https://island.io).

### CODE REVIEW

File: docs/guides/island-integration.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+---
+
+[Island](https://www.island.io/) is an enterprise-grade browser, offering a
+Chromium-based experience similar to popular web browsers like Chrome and Edge.
+It includes built-in security features for corporate applications and data,
+aiming to bridge the gap between consumer-focused browsers and the security
+needs of the enterprise.
+
+Coder natively integrates with Island's feature set, which include data loss
+protection (DLP), application awareness, browser session recording, and single
```
Comment (mattvollmer) [inline docs/guides/island-integration.md:20]: "data protection" per the PMM @ Island

File: docs/guides/island-integration.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+### Create an Application Group for Coder
+
+We recommend creating an Application Group specific to Coder in the Island
+Management console. This Application Group object will be referenced when
+creating browser policies.
+
+[See the Island documentation for creating an Application Group](https://documentation.island.io/docs/create-and-configure-an-application-group-object).
+
+## Advanced Data Loss Protection
```
Comment (mattvollmer) [inline docs/guides/island-integration.md:34]: Advanced Data Protection

**Comment by ericpaulsen** [comment]
recheck

**Comment by mattvollmer** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mattvollmer** [comment]
recheck

**Review by mattvollmer — APPROVED** [review]

**Review by SMWCoder — APPROVED** [review]
I have reviewed the changes

File: docs/guides/island-integration.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
+# Island Browser Integration
+
+<div>
+  <a href="https://github.com/ericpaulsen" style="text-decoration: none; color: inherit;">
+    <span style="vertical-align:middle;">M Atif Ali</span>
```
Comment (matifali) [inline docs/guides/island-integration.md:5]:
```suggestion
    <span style="vertical-align:middle;">Eric Paulson</span>
```

**Pull request merged**